### PR TITLE
Mobile/feature/navigation-drawer

### PR DESCRIPTION
**Pull request by YigitSekerci**
I have implemented the navigation drawer according [this issue](https://github.com/bounswe/bounswe2023group4/issues/224) and figma.

### CODE REVIEW

File: prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/common/NavigationDrawer.kt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+import com.bounswe.predictionpolls.ui.theme.PredictionPollsTheme
+import com.bounswe.predictionpolls.utils.NavItem
+import kotlinx.coroutines.launch
+
+@Composable
+fun NavigationDrawer(
+    modifier: Modifier = Modifier,
+    selectedNavItem: NavItem,
+    onButtonClick: (NavItem) -> Unit = {},
+    content: @Composable (() -> Unit) -> Unit = {}
```
Comment (aemresafak) [inline prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/common/NavigationDrawer.kt:46]: it is hard for me to understand what does the lambda in the composable lambda mean? What is this lambda? Can we name it?

File: prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/utils/NavItem.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.bounswe.predictionpolls.utils
```
Comment (aemresafak) [inline prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/utils/NavItem.kt:1]: sealed class DrawerItem(val route: String, @StringRes val resourceId: Int, @DrawableRes val iconId: Int) {
    object Profile : DrawerItem("profile", R.string.nav_drawer_profile, R.drawable.ic_profile)
    object Feed : DrawerItem("feed", R.string.nav_drawer_feed, R.drawable.ic_feed)
    object VotePoll : DrawerItem("vote", R.string.nav_drawer_vote, R.drawable.ic_vote)
    object CreatePoll : DrawerItem("create_poll", R.string.nav_drawer_create, R.drawable.ic_create)
    object Moderation : DrawerItem("moderation", R.string.nav_drawer_moderation, R.drawable.ic_moderation)
    object Leaderboard : DrawerItem("leaderboard", R.string.nav_drawer_leaderboard, R.drawable.ic_leaderboard)
    object Notifications : DrawerItem("notifications", R.string.nav_drawer_notifications, R.drawable.ic_notifications)
}

Seems like a better solution to me. Better naming, and more concise

**Review by aemresafak — CHANGES_REQUESTED** [review]

File: prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/utils/NavItem.kt (hunk 2 above)
Comment (YigitSekerci) [reply prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/utils/NavItem.kt:1]: I want to use this Item also in main navigation so to the name is not suitable however approach is better I will switch to this.

File: prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/common/NavigationDrawer.kt (hunk 1 above)
Comment (YigitSekerci) [reply prediction-polls/android/app/src/main/java/com/bounswe/predictionpolls/common/NavigationDrawer.kt]: You're right my mistake. I fixed it.

**Comment by YigitSekerci** [comment]
I have implemented the changes can you review it again? @aemresafak

**Review by ahmet-emre-bundle — APPROVED** [review]

**Pull request merged**